File: articles/reading-notes/head-first-design-pattern/code/02-obsever-pattern/02-NSNoticefaction/02-NSNoticefaction/ViewController.m
//
//  ViewController.m
//  02-NSNoticefaction
//
//  Created by zhouke on 2018/6/21.
//  Copyright © 2018年 zhouke. All rights reserved.
//

#import "ViewController.h"

#define NOTIFICATION_NAME @"NSNotificationName"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(noticeAction) name:NOTIFICATION_NAME object:nil];
}

- (void)noticeAction
{
    NSLog(@"getNotice");
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [[NSNotificationCenter defaultCenter] postNotificationName:NOTIFICATION_NAME object:nil userInfo:nil];
}


@end

File: docs/articles/reading-notes/Graphic-Design-Patterns/code/07-State/07-State/ViewController.swift
//
//  ViewController.swift
//  07-State
//
//  Created by zhouke on 2018/11/2.
//  Copyright © 2018 zhouke. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let context = Context(originalState: ConcreteStateA.share())
        
        context.request()
        context.request()
        context.request()
        
    }


}

